extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/tracing/client/QuarkusRestClientListener.java

```diff
@@ -0,0 +1,25 @@
+package io.quarkus.opentelemetry.tracing.client;
+
+import javax.enterprise.inject.Instance;
+import javax.enterprise.inject.spi.CDI;
+
+import org.eclipse.microprofile.rest.client.RestClientBuilder;
+import org.eclipse.microprofile.rest.client.spi.RestClientListener;
+import org.jboss.logging.Logger;
+
+import io.opentelemetry.api.trace.Tracer;
+
+public class QuarkusRestClientListener implements RestClientListener {
+    private static final Logger log = Logger.getLogger(QuarkusRestClientListener.class);
+
+    @Override
+    public void onNewClient(Class<?> clientInterface, RestClientBuilder restClientBuilder) {
+        Instance<Tracer> tracerInstance = CDI.current().select(Tracer.class);
+
+        if (tracerInstance.isResolvable()) {
+            restClientBuilder.register(new ClientTracingFilter(tracerInstance.get()));
+        } else {
+            log.warn("Unable to resolve Tracer instance, no ClientTracingFilter registered on REST Client.");
+        }
+    }
```

Comment:
If Tracer is application scoped, you should be able to cache the lookup so you don't resolve this every time a rest client is created. Especially if the ClientTracingFilter is reusable, you can make this a lazy, more-or-less-single instance (e.g. if you get more than one, it's ok .. so you don't have to go nuts with concurrency)